### PR TITLE
User list endpoint returns current user only instead of error when permission is missing

### PR DESCRIPTION
**Pull request by janheise**
## Description

We are using the `list` method on the user resource in a context where a user can only have the Reader role. That role lacks the permission to list all users. In this case, we only want to return the user itself in the list.

Fixes https://github.com/Graylog2/graylog-plugin-enterprise/issues/8692

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java [hunk 1]
```diff
@@ -233,15 +234,27 @@ private UserSummary returnSummary(UserContext userContext, User user) {
...
-        return listUsersSelective(includePermissions, includeSessions);
+            @ApiParam(name = "include_sessions") @QueryParam("include_sessions") @DefaultValue("true") boolean includeSessions,
+            @Context SearchUser searchUser) {
+        return searchUser.isPermitted(RestPermissions.USERS_LIST) ?
+                listUsersSelective(includePermissions, includeSessions) :
+                listForLoggedInUser(searchUser, includePermissions, includeSessions);
+    }
+
+    private UserList listForLoggedInUser(final SearchUser searchUser, final boolean includePermissions, final boolean includeSessions) {
+        final var user = userManagementService.load(searchUser.username());
```
Comment (dennisoelkers) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java:248]: Is there a reason why you are not just using `searchUser.getUser()` here?

File: graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java [hunk 2]
```diff
@@ -233,15 +234,27 @@ private UserSummary returnSummary(UserContext userContext, User user) {
...
+        final Optional<AllUserSessions> optSessions = includeSessions ? Optional.of(AllUserSessions.create(sessionService)) : Optional.empty();
+        if(user != null) {
```
Comment (dennisoelkers) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java:250]: In which cases would this be `null`?

**Review by dennisoelkers — COMMENTED** [review]
Added some questions inline.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java (hunk 1 above)
Comment (janheise) [reply graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java]: no, fixed it

File: graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java (hunk 2 above)
Comment (janheise) [reply graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java]: fixed it

File: changelog/unreleased/pr-20558.toml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+type = "c" # One of: a(dded), c(hanged), d(eprecated), r(emoved), f(ixed), s(ecurity)
+message = "UserResource.list() now works for every authenticated user. In case the calling user lacks the 'RestPermissions.USERS_LIST' permission, only the calling user is returned."
```
Comment (dennisoelkers) [inline changelog/unreleased/pr-20558.toml:2]: I think we should write this in a more user-friendly way (changelogs are not meant for developers, but for users/people consuming our API). A suggestion:
```suggestion
message = "API endpoint for listing users will not return an error anymore when user is lacking permission, but return current user only instead."
```

**Review by dennisoelkers — CHANGES_REQUESTED** [review]
Suggestion for making changelog snippet simpler

**Review by dennisoelkers — APPROVED** [review]

**Pull request merged**